lib/codecs/XsdDateEncoder.js

```diff
@@ -5,14 +5,21 @@ import {Token, Type} from 'cborg';
 import {CborldEncoder} from './CborldEncoder.js';
 
 export class XsdDateEncoder extends CborldEncoder {
-  constructor({value, parsed} = {}) {
+  constructor({value, parsed, compressionMap} = {}) {
     super();
     this.value = value;
     this.parsed = parsed;
+    this.compressionMap = compressionMap;
   }
 
   encode() {
-    const {value, parsed} = this;
+    const {value, parsed, compressionMap} = this;
+
+    //use string table if possible
+    if(compressionMap.has(value)) {
+      return compressionMap.get(value);
+    }
```

Comment:
Also... doesn't this need to return a `Token` instance? And this seems like it would cause a conflict :/. The regular compression here uses a uint type already... I'd think we'd need to use a byte array or something here instead like we do in that other place, so this is distinguishable. Either that -- or say that you can't replace specific dates using the string table.

```suggestion
    const {value, parsed, id} = this;

    // use `id` from string table if possible
    if(id !== undefined) {
      return new Token(Type.uint, value);
    }
```